src/components/label-pill/LabelPill.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import Tooltip from '../tooltip';
+
+import './LabelPill.scss';
+
+// TODO: Unify notification types `InlineNoticeType` from core
+type Props = {
+    children: React.Node,
+    tooltip?: string,
+    type?: 'info' | 'warning' | 'highlight' | 'error' | 'alert' | 'success',
+};
+
+const LabelPill = ({ children, type = 'default', tooltip = '', ...rest }: Props) => {
+    React.Children.only(children);
+    const contentType = React.Children.map(children, child => child.type.name)[0];
+    const labelPill = (
+        <div
+            className={classNames(classNames('bdl-LabelPill', type !== 'default' ? `bdl-LabelPill--${type}` : ''))}
+            {...rest}
+        >
+            <div
+                className={classNames(
+                    'bdl-LabelPill-content',
+                    contentType.indexOf('Formatted') === 0 ? 'bdl-LabelPill-has-text' : 'bdl-LabelPill-has-component',
```

Comment:
I dont think this should be done. Someone can use intl.format too.

Reply:
any ideas on how to tell whether there's a text node or an icon in the -content?

Reply:
You don't you make two different components to choose from

Reply:
actually, i'll go with a less clever approach here to see how that looks

Reply:
You could potentially move all this under the pill folder.
Make your component PillBase, and then export out two components - LabelPill, IconPill or something like that. And when we consolidate, LabelPill can replace the existing Pill.js that used inside pill selector

Reply:
If it's ok, will postpone this until we gather up all the pill-like things, and evaluate them. maybe there's an intermediate step that would conflict with this idea